protocols/v2/roles-logic-sv2/src/job_creator.rs

```diff
@@ -49,6 +76,11 @@ impl JobCreator {
         let bip34_len = script_prefix[1] as usize;
         let bip34_bytes = script_prefix[1..2 + bip34_len].to_vec();
 
+        // NOTE: There maybe be better ways to determine if the NewTemplate
+        // has segwit inputs. In this case, we are assuming extra coinbase_tx_outputs
+        // means a witness commitment.
+        let segwit_flag = new_template.coinbase_tx_outputs_count > 0;
```

Comment:
we can use `new_template.witness_reserve_value`

Reply:
@ccdle12 Told me this requires us to review https://github.com/stratum-mining/sv2-spec/pull/15 first, can we do that @Fi3 ?